src/modules/java.rs

```diff
@@ -0,0 +1,77 @@
+use std::process::Command;
+
+use ansi_term::Color;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Java version
+///
+/// Will display the Java version if any of the following criteria are met:
+///     - Current directory contains a file with a `.java` extension
+///     - Current directory contains a `pom.xml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_java_project = context
+        .try_begin_scan()?
+        .set_files(&["pom.xml", "build.gradle"])
+        .set_extensions(&["java", "class", "jar"])
+        .is_match();
+
+    if !is_java_project {
+        return None;
+    }
+
+    match get_java_version() {
+        Some(java_version) => {
+            const JAVA_CHAR: &str = "☕ ";
+
+            let mut module = context.new_module("java");
+            let module_style = module
+                .config_value_style("style")
+                .unwrap_or_else(|| Color::Red.dimmed());
+            module.set_style(module_style);
+
+            let formatted_version = format_java_version(java_version)?;
+            module.new_segment("symbol", JAVA_CHAR);
+            module.new_segment("version", &formatted_version);
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_java_version() -> Option<String> {
+    match Command::new("java").arg("-Xinternalversion").output() {
```

Comment:
Should we take into account the `JAVA_HOME` variable?
If exists  the version could be `$JAVA_HOME/bin/java -Xinternalversion`. 
When using maven it will take priority over the java binary in the `PATH`.

Reply:
I like this idea. I'm not a fan of relying on JAVA_HOME being set (simply because half the envars that really ought to be set usually aren't), but I think checking for the existence of a JAVA_HOME variable should be more than fast enough.

Do you know if that variable gets set on Windows as well, or is that a UNIX-like-only thing?

Reply:
Answer to my own question: it seems like (as with many envars) [it's supposed to be set on windows, but a lot of installers fail to do so](https://stackoverflow.com/questions/2619584/how-to-set-java-home-on-windows-7).

Reply:
I don't think it's necessary to check `$JAVA_HOME`. Running `$JAVA_HOME/bin/java -Xinternalversion` should be identical to running `java -Xinternalversion`. If `java` isn't on the `PATH` environment variable, then Java simply isn't set up properly.

Reply:
https://www.baeldung.com/maven-java-home-jdk-jre

First checks `$JAVA_HOME`, otherwise defaults to `java` on the `$PATH`.

But I've never had to set up separate Java installations and setting up `$JAVA_HOME` for different projects...

So maybe we should follow the same strategy as Maven:

Check for `$JAVA_HOME`.
Otherwise just use the `java` that is on the `$PATH`.